src/pybind/mgr/dashboard/controllers/__init__.py

```diff
@@ -705,21 +705,23 @@ def inner(*args, **kwargs):
             kwargs.update(params)
 
             if version is not None:
+                exp_version = APIVersion.to_string(version)
                 accept_header = cherrypy.request.headers.get('Accept')
-                if accept_header and accept_header.startswith('application/vnd.ceph.api.v'):
-                    req_match = re.search(r"\d\.\d", accept_header)
-                    if req_match:
-                        req_version = req_match[0]
+                req_match = APIVersion.from_string(accept_header)
+                if accept_header and req_match:
+                    req_version = req_match
```

Comment:
Don't shortcut variables, it breaks the magic of Pythonic code. Exp means expected? Req, required, requested?

```py
                server_version = APIVersion._make(version) # assuming version is tuple, not str
                client_version = APIVersion.from_string(cherrypy.request.headers.get('Accept'))
                try:
                  if client_version.handled_by(server_version):
                    ...  
                except AttributeError: # None.handled_by(...)
```

Reply:
Will change vars naming.

Reply:
In this very case it's better to implement a `handles` or `supports` method:
```py
class APIVersion():
  ...
  def supports(self, version):
    if version is None:
      raise cherrypy.HTTPError(415, "Unable to find version in request header")
    if self.major == version.major ...:
      raise cherrypy.HTTPError(415, "Incorrect version: ...")


...
if server_version.supports(client_version):
  ...
```